### PR TITLE
Rust toolchain upgrade to 1.64.0

### PR DESCRIPTION
**Pull request by bchalios**
## Changes

Upgrade Rust toolchain to latest stable (1.64.0). This includes several fixes:

- New bindings for v5.10 kernel
- `cargo clippy` and `cargo fmt` fixes
- Fixes in tests due to changes in the toolchain
- Pylint fixes
- Updated performance baselines

## Reason

Rust toolchain we were using was quite old and included a known security (which was not affecting us).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] New `unsafe` code is documented.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

**Review by zulinx86 — CHANGES_REQUESTED** [review]
The following three commits are displayed as "Unverified". All commits should be signed.
- [Add script to generate bindings](https://github.com/firecracker-microvm/firecracker/pull/3176/commits/90e2cba80c68e7e2ec7cd21bc152dac773d4bf73) 
- [Regenerate bindings against v5.10](https://github.com/firecracker-microvm/firecracker/pull/3176/commits/a18ef635932f29a7d5db7cc8d725cf3d6b891821) 
- [Update coverage value](https://github.com/firecracker-microvm/firecracker/pull/3176/commits/b80f8937f11f882ed5c927764015a2b0fbdacbfc)

**Comment by bchalios** [comment]
> The following three commits are displayed as "Unverified". All commits should be signed.
> 
>     * [Add script to generate bindings](https://github.com/firecracker-microvm/firecracker/pull/3176/commits/90e2cba80c68e7e2ec7cd21bc152dac773d4bf73)
> 
>     * [Regenerate bindings against v5.10](https://github.com/firecracker-microvm/firecracker/pull/3176/commits/a18ef635932f29a7d5db7cc8d725cf3d6b891821)
> 
>     * [Update coverage value](https://github.com/firecracker-microvm/firecracker/pull/3176/commits/b80f8937f11f882ed5c927764015a2b0fbdacbfc)

The requirement is that commits are signed with `git commit -s` (emphasis on the lower-case `s`) not with `-S` which signs them with a GPG key. The lack of later is what makes commits appear as "Unverified"

**Review by dianpopa — DISMISSED** [review]

**Review by mattschlebusch — DISMISSED** [review]
Overall looks good to me as most changes are syntactic sugar. The only question mark to is centred around performance thresholds, but they aren't large enough to warrant blocking the PR.

File: src/arch/src/x86_64/regs.rs [hunk 1]
```diff
@@ -77,8 +77,7 @@ pub fn setup_fpu(vcpu: &VcpuFd) -> std::result::Result<(), SetupFpuError> {
         ..Default::default()
     };
 
-    let res = vcpu.set_fpu(&fpu)?;
-    Ok(res)
+    vcpu.set_fpu(&fpu).map_err(SetupFpuError)
```
[the remark this answers is not in the document]
Comment (bchalios) [reply src/arch/src/x86_64/regs.rs:80]: So the change here is about coding style. I _think_ what you say make sense, but I'd prefer not to do that change as part of the Rust upgrade. It looks a bit out of scope.

File: src/cpuid/src/transformer/intel.rs [hunk 2]
```diff
@@ -181,7 +181,7 @@ mod tests {
             padding: [0, 0, 0],
         };
 
-        assert!(update_deterministic_cache_entry(&mut entry, &vm_spec).is_ok());
+        assert!(update_deterministic_cache_entry(entry, &vm_spec).is_ok());
```
[the remark this answers is not in the document]
Comment (bchalios) [reply src/cpuid/src/transformer/intel.rs:184]: I'm not sure I understand the comment. Could you give a bit more context?

[the remark this answers is not in the document]
Comment (JonathanWoollett-Light) [reply src/cpuid/src/transformer/intel.rs:184]: `matches!` is preferable to `is_ok()`. So in instances where it was relatively easy and connected to required changes I made this change.

File: resources/seccomp/aarch64-unknown-linux-musl.json [hunk 3]
```diff
@@ -170,6 +170,19 @@
...
+            {
+                "syscall": "futex",
+                "comment": "Used for synchronization",
+                "args": [
+                    {
+                        "index": 1,
+                        "type": "dword",
+                        "op": "eq",
+                        "val": 137,
+                        "comment": "FUTEX_WAIT_BITSET_PRIVATE"
```
[the remark this answers is not in the document]
Comment (bchalios) [reply resources/seccomp/aarch64-unknown-linux-musl.json:182]: It is still used. Removing it from the allow-list breaks stuff.

File: src/cpuid/src/transformer/intel.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattschlebusch) [reply src/cpuid/src/transformer/intel.rs:184]: My point is that there are a number of places where `matches!` could be used but isn't. There are some lines that make the refactor, but some that don't.

[the remark this answers is not in the document]
Comment (bchalios) [reply src/cpuid/src/transformer/intel.rs:184]: oh, `matches!` is useful when you want to check for a pattern. for staff that are easily comparable, or return just one value, `assert` and `assert_eq` are just fine. I think that's why clippy changed some and no others.

**Pull request closed without merging**